crates/modelardb_storage/src/delta_lake.rs

```diff
@@ -0,0 +1,495 @@
+            ("aws_access_key_id".to_owned(), access_key_id),
+            ("aws_secret_access_key".to_owned(), secret_access_key),
+            ("aws_endpoint_url".to_owned(), endpoint),
+            ("aws_bucket_name".to_owned(), bucket_name),
+            ("aws_s3_allow_unsafe_rename".to_owned(), "true".to_owned()),
+        ]);
+
+        let url = Url::parse(&location)
+            .map_err(|error| ModelarDbStorageError::InvalidArgument(error.to_string()))?;
+
+        // Build the Amazon S3 object store with the given storage options manually to allow http.
+        let object_store = storage_options
+            .iter()
+            .fold(
+                AmazonS3Builder::new()
+                    .with_url(url.to_string())
+                    .with_allow_http(true),
+                |builder, (key, value)| match key.parse() {
+                    Ok(k) => builder.with_config(k, value),
+                    Err(_) => builder,
+                },
+            )
+            .build()?;
+
+        Ok(DeltaLake {
+            location,
+            storage_options,
+            object_store: Arc::new(object_store),
+            maybe_local_file_system: None,
+        })
+    }
+
+    /// Create a new [`DeltaLake`] that manages the Delta tables in an object store with an
+    /// Azure-compatible API. Returns a [`ModelarDbStorageError`] if a connection to the object store
+    /// cannot be made.
+    pub fn try_from_azure_configuration(
+        account_name: String,
+        access_key: String,
+        container_name: String,
+    ) -> Result<Self> {
+        let location = format!("az://{container_name}");
+
+        // TODO: Needs to be tested.
+        let storage_options = HashMap::from([
+            ("azure_storage_account_name".to_owned(), account_name),
+            ("azure_storage_account_key".to_owned(), access_key),
+            ("azure_container_name".to_owned(), container_name),
+        ]);
+        let url = Url::parse(&location)
+            .map_err(|error| ModelarDbStorageError::InvalidArgument(error.to_string()))?;
+        let (object_store, _path) = object_store::parse_url_opts(&url, &storage_options)?;
+
+        Ok(DeltaLake {
+            location,
+            storage_options,
+            object_store: Arc::new(object_store),
+            maybe_local_file_system: None,
+        })
+    }
+
+    /// Return an [`ObjectStore`] to access the root of the Delta Lake.
+    pub fn object_store(&self) -> Arc<dyn ObjectStore> {
+        self.object_store.clone()
+    }
+
+    /// Return a [`LocalFileSystem`] to access the root of the Delta Lake if it uses a local data
+    /// folder.
+    pub fn local_file_system(&self) -> Option<Arc<LocalFileSystem>> {
+        self.maybe_local_file_system.clone()
+    }
+
+    /// Return a [`DeltaTable`] for manipulating the metadata table with `table_name` in the
+    /// Delta Lake, or a [`ModelarDbStorageError`] if a connection cannot be established or the
```

Comment:
Added "to the delta lake" to each doc comment to make it more clear. It is since `storage_options` could refer to a remote object store which could result in connection issues.